src/Wallet.purs

```diff
@@ -29,127 +30,51 @@ import Effect (Effect)
 import Effect.Aff (Aff)
 import Effect.Class (liftEffect)
 import Effect.Exception (throw)
-import Effect.Ref as Ref
 import FfiHelpers (MaybeFfiHelper, maybeFfiHelper)
 import Helpers ((<<>>))
 import Serialization as Serialization
 import Serialization.Address (Address, addressFromBytes)
 import Types.ByteArray (byteArrayToHex)
-import Types.CborBytes
-  ( CborBytes
-  , cborBytesToHex
-  , rawBytesAsCborBytes
-  )
+import Types.CborBytes (CborBytes, cborBytesToHex, rawBytesAsCborBytes)
 import Types.RawBytes (RawBytes, hexToRawBytes)
-import Types.Transaction (TransactionHash(TransactionHash))
 import Untagged.Union (asOneOf)
 
--- At the moment, we only support Nami's wallet. In the future we will expand
--- this with more constructors to represent out-of-browser wallets (e.g. WBE)
-data Wallet = Nami NamiWallet
-
--------------------------------------------------------------------------------
--- Nami backend
--------------------------------------------------------------------------------
--- Record-of-functions for real or mocked Nami wallet, includes `Ref` to
--- connection (e.g. with `window.cardano.nami` as a `NamiConnection`)
-type NamiWallet =
-  { -- A reference to a connection with Nami, i.e. `window.cardano.nami`
-    connection :: Ref.Ref NamiConnection
+type Cip30Wallet =
+  { -- A reference to a connection with the wallet, i.e. `window.cardano.nami`
+    connection :: Cip30Connection
   -- Get the address associated with the wallet (Nami does not support
   -- multiple addresses)
-  , getWalletAddress :: NamiConnection -> Aff (Maybe Address)
+  , getWalletAddress :: Cip30Connection -> Aff (Maybe Address)
   -- Get the collateral UTxO associated with the Nami wallet
-  , getCollateral :: NamiConnection -> Aff (Maybe TransactionUnspentOutput)
+  , getCollateral :: Cip30Connection -> Aff (Maybe TransactionUnspentOutput)
   -- Sign a transaction with the current wallet
-  , signTx :: NamiConnection -> Transaction -> Aff (Maybe Transaction)
+  , signTx :: Cip30Connection -> Transaction -> Aff (Maybe Transaction)
   -- Sign a transaction with the current wallet
```

Comment:
Not related to the current changes but this comment should be made more accurate